ide/textmate.lexer/arch.xml

```diff
@@ -0,0 +1,1115 @@
+            If possible please provide simple diagrams.
+            </hint>
+        </question>
+-->
+ <answer id="arch-overall">
+  <p>
+   This is a lexer an syntax highlighting for TextMate grammars. The grammars
+   may be regitered using the system layer.
+   <api name="register-grammar" group="layer" type="export" category="devel" />
+  </p>
+ </answer>
+
+
+
+<!--
+        <question id="arch-quality" when="init">
+            How will the <a href="http://www.netbeans.org/community/guidelines/q-evangelism.html">quality</a>
+            of your code be tested and 
+            how are future regressions going to be prevented?
+            <hint>
+            What kind of testing do
+            you want to use? How much functionality, in which areas,
+            should be covered by the tests? How you find out that your
+            project was successful?
+            </hint>
+        </question>
+-->
+ <answer id="arch-quality">
+  <p>
+   Unit tests for the lexer.
+  </p>
+ </answer>
+
+
+
+<!--
+        <question id="arch-time" when="init">
+            What are the time estimates of the work?
+            <hint>
+            Please express your estimates of how long the design, implementation,
+            stabilization are likely to last. How many people will be needed to
+            implement this and what is the expected milestone by which the work should be 
+            ready?
+            </hint>
+        </question>
+-->
+ <answer id="arch-time">
+  <p>
+   Unclear.
+  </p>
+ </answer>
+
+
+
+<!--
+        <question id="arch-usecases" when="init">
+            <hint>
+                Content of this answer will be displayed as part of page at
+                http://www.netbeans.org/download/dev/javadoc/usecases.html 
+                You can use tags &lt;usecase name="name&gt; regular html description &lt;/usecase&gt;
+                and if you want to use an URL you can prefix if with @TOP@ to begin
+                at the root of your javadoc
+            </hint>
+        
+            Describe the main <a href="http://wiki.netbeans.org/API_Design#The_Importance_of_Being_Use_Case_Oriented">
+            use cases</a> of the new API. Who will use it under
+            what circumstances? What kind of code would typically need to be written
+            to use the module?
+        </question>
+-->
+ <answer id="arch-usecases">
+  <p>
+      <usecase id="register-grammar" name="Register grammar">
+          To register a new TextMate grammar, add it as a file into the system filesystem,
+          under "Editors/&lt;mime-type&gt;/", and make sure it has an attribute "textmate-grammar",&lt;/mime-type&gt;,
+          whose value is a string representing the TextMate top-level scope. For example:
+          <pre>
+    &lt;folder name="Editors"&gt;
+        &lt;folder name="text"&gt;
+            &lt;folder name="x-kotlin"&gt;
+                &lt;file name="Kotlin.tmLanguage.json" url="Kotlin.tmLanguage.json"&gt;
```

Comment:
It is not clear to me where I should download `Kotlin.tmLanguage.json` from.

Reply:
Not sure what you mean - if someone wants to integrate a grammar to NetBeans, they have the grammar, and this example shows them how to integrate them? Should the arch.xml list all known locations where the grammars could potentially be found?